backend/command/src/items/interfaces/insert-item.dto.ts

```diff
@@ -6,55 +6,90 @@ import {
   IsNotEmpty,
   IsNumber,
   IsOptional,
+  IsPositive,
   IsString,
   ValidateNested,
 } from 'class-validator';
+import { IsBiggerThan } from '../../shared/validators/is-bigger-than';
 
 class Weight {
   @IsNumber()
-  @ApiProperty({ default: '1.234e100' })
+  @IsPositive()
+  @ApiProperty({
+    exclusiveMinimum: true,
+    minimum: 0,
+    description: 'Value of weight in gram.',
+    example: 150,
+  })
   value: number;
 
   @IsBoolean()
   @IsOptional()
-  @ApiPropertyOptional({ default: false })
+  @ApiPropertyOptional({
+    default: false,
+    description: 'Set whether the weight is an approximate weight.',
+    example: true,
+  })
   isCa?: boolean;
 
   @IsNumber()
+  @IsPositive()
   @IsOptional()
-  @ApiPropertyOptional()
+  @IsBiggerThan('value', {
+    message: 'additionalValue has to be bigger than value',
+  })
+  @ApiPropertyOptional({
+    exclusiveMinimum: true,
+    minimum: 0,
+    description:
+      'Additional value of weight in gramm. Has to be bigger than value. Use this if you have a range weight.',
+    example: 250,
+  })
   additionalValue?: number;
 }
 
 export class InsertItemDto {
   // TODO: Slug here or on event write
   @IsString()
   @IsNotEmpty()
-  @ApiProperty()
+  @ApiProperty({ description: 'Name of the item.', example: 'Apple' })
   name: string;
 
   @ValidateNested()
-  @ApiProperty()
   @Type(() => Weight)
+  @ApiProperty({
+    type: Weight,
+    description: 'Weight of the item.',
+    example: { value: 150, isCa: false, additionalValue: 250 },
+  })
   weight: Weight;
 
   @IsArray()
   @IsOptional()
-  @ApiPropertyOptional({ type: [String] })
-  @Type(() => String)
+  @IsString({ each: true })
+  @ApiPropertyOptional({
+    type: [String],
+    description: 'Tags of the item.',
+    example: ['fruit', 'red'],
+  })
   tags?: string[];
 
   @IsString()
   @IsOptional()
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({ description: 'Image data' })
```

Comment:
Isnt this just the image URL?

Reply:
I thought about this as well. We can do two things:

1. Safe the whole URL (https://image.wwweights.com/img/XXX.jpg)
2. Safe only the ID/hash (XXX)

The second one is more flexibel (we can simply change the URL / Domain) and also it's easier to send requests e.g. delete (we do not have to regex the URL). However at the current state we could also have image from other CDNs and domains so maybe we have to allow booth. That's the reason I just put in "Image data".